docs/smart/principals.md

```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
 
 ## Principals and tx-sender
 
-A principal is represented by a public-key hash or multi-signature Stacks address. Assets in Clarity and the Stacks blockchain are "owned" by objects of the principal type; put another way, principal object types may own an asset. 
+Assets in the smart contracting language and blockchain are
+"owned" by objects of the principal type, meaning that any object of
+the principal type may own an asset. For the case of public-key hash
+and multi-signature Stacks addresses, a given principal can operate on
+their assets by issuing a signed transaction on the blockchain. _Smart
+contracts_ may also be principals (reprepresented by the smart
+contract's identifier), however, there is no private key associated
+with the smart contract, and it cannot broadcast a signed transaction
+on the blockchain.
+
+A Clarity contract can use a globally defined `tx-sender` variable to
+obtain the current principal. The following example defines a transaction
+type that transfers `amount` uSTX from the sender to a recipient if some
+condition is met, otherwise returning a 400 error code.
 
-A given principal operates on its assets by issuing a signed transaction on the Stacks blockchain. A Clarity contract can use a globally defined `tx-sender` variable to obtain the current principal.
-
-The following user-defined function transfers an asset, in this case, tokens, between two principals:
-
-```
-(define (transfer! (sender principal) (recipient principal) (amount int))
-  (if (and
-        (not (eq? sender recipient))
-        (debit-balance! sender amount)
-        (credit-balance! recipient amount))
-    true
-    false))
+```cl
```

Comment:
scheme?

Reply:
Fixed in 1a0d04d

Reply:
I have found using triple-tick `clojure` to result in the most fitting formatting.